### PR TITLE
coprocessor/xeval: add abs real function impl

### PR DESCRIPTION
**Pull request by bailaohe**
implementation of builtin abs_real in coprocessor/xeval

### CODE REVIEW

**Comment by siddontang** [comment]
Thanks @bailaohe 

Please sign the CLA.

**Comment by siddontang** [comment]
PTAL @AndreMouche @andelf

File: src/coprocessor/xeval/builtin_math.rs [hunk 1]
```diff
@@ -39,9 +40,13 @@ impl Evaluator {
...
 
-    pub fn abs_real(&mut self, _ctx: &EvalContext, _expr: &Expr) -> Result<Datum> {
-        // TODO add impl
-        Err(Error::Eval(ERROR_UNIMPLEMENTED.to_owned()))
+    pub fn abs_real(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
+        let child = try!(self.get_one_child(expr));
+        let d = try!(self.eval(ctx, child));
+        match d {
+            Datum::F64(f) => Ok(Datum::F64(f.abs())),
+            _ => invalid_type_error(&d, TYPE_FLOAT),
```
Comment (AndreMouche) [inline src/coprocessor/xeval/builtin_math.rs]: It seems we should return `Datum::None` when d is `None` according tidb (https://github.com/pingcap/tidb/blob/master/expression/builtin_math.go#L121)  @XuHuaiyu

Comment (bailaohe) [inline src/coprocessor/xeval/builtin_math.rs]: thx, do you mean Datum::Null? 
I added a branch for it. PTAL.

**Comment by AndreMouche** [comment]
@BusyJay @andelf PTAL

**Comment by AndreMouche** [comment]
Thanks for your contribution and please fix the conflicts. @bailaohe

**Comment by bailaohe** [comment]
@BusyJay @AndreMouche please help me re-trigger the Travis-CI procedure. Thx. Building is blocked with some rocksdb issue.

**Comment by bailaohe** [comment]
@AndreMouche hi，I found the build process failed with rocksdb compile error. https://travis-ci.org/pingcap/tikv/jobs/256855637. May you address this and rerun the Travis-build? Thanks.

**Comment by AndreMouche** [comment]
We're pretty sorry to tell you that there is something wrong with our CI, and the related PR is on the way to merge(https://github.com/pingcap/tikv/pull/2074).  Thanks for your contribution again. @bailaohe

**Pull request merged**